cnd/src/http_api/routes/index.rs

```diff
@@ -127,20 +124,41 @@ pub struct Body<A, B> {
     pub role: Http<Role>,
 }
 
-impl From<Body<Herc20EthereumErc20, HalightLightningBitcoin>>
-    for Herc20HalightBitcoinCreateSwapParams
-{
+impl From<Body<Herc20EthereumErc20, HalightLightningBitcoin>> for swap_digest::Herc20Halight {
     fn from(body: Body<Herc20EthereumErc20, HalightLightningBitcoin>) -> Self {
         Self {
-            role: body.role.0,
-            peer: body.peer.into(),
-            ethereum_identity: body.alpha.identity,
             ethereum_absolute_expiry: body.alpha.absolute_expiry.into(),
-            ethereum_amount: body.alpha.amount,
-            lightning_identity: body.beta.identity,
+            erc20_amount: body.alpha.amount,
+            token_contract: body.alpha.contract_address,
             lightning_cltv_expiry: body.beta.cltv_expiry.into(),
             lightning_amount: body.beta.amount.0,
-            token_contract: body.alpha.contract_address,
+        }
+    }
+}
+
+trait ToCreatedSwap<A, B> {
```

Comment:
Are we using this trait somewhere for an abstraction? If no, then this fn can just go on the `Body` type.

Reply:
This will be used as soon as we add the other direction for this trading pair.  Since touching `index.rs` is already dangerously close to out of scope for this PR not adding this now.
```

impl ToCreatedSwap<halight::CreatedSwap, herc20::CreatedSwap>
    for Body<HalightLightningBitcoin, Herc20EthereumErc20>
{
    fn to_created_swap(
        &self,
        swap_id: LocalSwapId,
    ) -> CreatedSwap<halight::CreatedSwap, herc20::CreatedSwap> {
        let body = self.clone();

        let alpha = halight::CreatedSwap::from(body.alpha);
        let beta = herc20::CreatedSwap::from(body.beta);

        CreatedSwap {
            swap_id,
            alpha,
            beta,
            peer: body.peer.into(),
            address_hint: None,
            role: body.role.0,
        }
    }
}
```

Reply:
How is this different from an `impl Body<halight::CreatedSwap, ...>`?

Reply:
Oh I did not catch that idea fully when you first suggested it, will implement as above.